### PR TITLE
diopi support profiler

### PR DESCRIPTION
**Pull request by caikun-pjlab**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation and Context

support diopi profiler to get more detail infomation

## Description

support diopi profiler to get more detail infomation

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/torch/functions.cpp [hunk 1]
```diff
@@ -193,8 +193,11 @@ diopiError_t diopiConvolution2d(diopiContextHandle_t ctx, diopiTensorHandle_t ou
     auto atStride = impl::aten::buildAtIntArray(stride);
     auto atPadding = impl::aten::buildAtIntArray(padding);
     auto atDilation = impl::aten::buildAtIntArray(dilation);
+    diopiRecordFunctionHandle_t record_function = nullptr;
```
Comment (yangbofun) [inline impl/torch/functions.cpp:196]: 这个record_function,设计时，可以删掉么，只有diopiRecordFunctionStart 和 diopiRecordFunctionEnd.

File: impl/torch/functions.cpp [hunk 2]
```diff
@@ -193,8 +193,11 @@ diopiError_t diopiConvolution2d(diopiContextHandle_t ctx, diopiTensorHandle_t ou
...
+    diopiRecordFunctionStart("at::diopi::call::convolution_out", &record_function);
```
Comment (yangbofun) [inline impl/torch/functions.cpp:197]: 这里写最原始的调用接口就行，at:: at::convolution_out

File: proto/include/diopi/diopirt.h [hunk 3]
```diff
@@ -103,6 +103,10 @@ struct diopiGenerator;
 typedef struct diopiGenerator* diopiGeneratorHandle_t;
 typedef const struct diopiGenerator* diopiConstGeneratorHandle_t;
 
+struct diopiRecordFunction;
+typedef struct diopiRecordFunction* diopiRecordFunctionHandle_t;
+typedef const struct diopiRecordFunction* diopiConstRecordFunctionHandle_t;
```
Comment (yangbofun) [inline proto/include/diopi/diopirt.h:108]: const的可以删了。

File: impl/torch/functions.cpp (hunk 2 above)
Comment (caikun-pjlab) [reply impl/torch/functions.cpp]: 已修改

File: proto/include/diopi/diopirt.h (hunk 3 above)
Comment (caikun-pjlab) [reply proto/include/diopi/diopirt.h]: 已删除，现在profiler这块都使用void*

File: impl/torch/functions.cpp (hunk 1 above)
Comment (caikun-pjlab) [reply impl/torch/functions.cpp]: 看起来不行，每次profiler需要有一个实体来控制开始和结束

**Review by yangbofun — APPROVED** [review]

**Review by zhaoguochun1995 — APPROVED** [review]

**Pull request merged**